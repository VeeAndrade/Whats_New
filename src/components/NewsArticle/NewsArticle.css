.article-section {
  display: flex;
  flex-flow: column nowrap;
  width: 90%;
  height: 430px;
}

.article-info {
  width: 100%;
  height: 200px;
  background-color: #fff;
  overflow: scroll;
}

.article-headline {
  margin-top: 5px;
  font-size: 1.3em;
  color: #000;
  height: 50%;
}

.article-description {
  margin: 0px 8px;
  color: #000;
  text-align: center;
  height: 50%;
}

.article-image {
  width: 100%;
  height: 40%;
  background-position: top;
  object-fit: cover;
  overflow: hidden;
}


.url-section {
  display: flex;
  width: 100%;
  background-color: #B56AF1;
  justify-content: space-between;
}

.article-url {
  font-size: 1.4em;
  color: #fff;
  width: 50%;
  text-decoration: none;
}

.article-url-arrow {
  width: 50%;
  font-size: 1.6em;
  color: #fff;
}
