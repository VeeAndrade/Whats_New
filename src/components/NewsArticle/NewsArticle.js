import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({headline, image, url, description}) => {
  return <section className='article-section'>
    <img className='article-image' src={image} alt='news'></img>
    <section className='article-info'>
      <h2 className='article-headline'>{headline}</h2>
      <p className='article-description'>{description}</p>
    </section>
    <section className='url-section'>
      <a className='article-url' href={url}>Link To article</a>
      <p className='article-url-arrow'>â†’</p>
    </section>
  </section>
}

export default NewsArticle;