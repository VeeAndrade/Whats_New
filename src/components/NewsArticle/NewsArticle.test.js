import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle'

describe('NewsArticle', () => {
  it('should render a new news article', () => {
    const wrapper = shallow(<NewsArticle 
    headline='Fake News'
    image='url@url.com'
    url='url@url.com'
    description='blah'
    />)

    expect(wrapper).toMatchSnapshot();
  });
});