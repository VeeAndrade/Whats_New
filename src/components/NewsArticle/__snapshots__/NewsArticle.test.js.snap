// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should render a new news article 1`] = `
<section
  className="article-section"
>
  <img
    alt="news"
    className="article-image"
    src="url@url.com"
  />
  <section
    className="article-info"
  >
    <h2
      className="article-headline"
    >
      Fake News
    </h2>
    <p
      className="article-description"
    >
      blah
    </p>
  </section>
  <section
    className="url-section"
  >
    <a
      className="article-url"
      href="url@url.com"
    >
      Link To article
    </a>
    <p
      className="article-url-arrow"
    >
      â†’
    </p>
  </section>
</section>
`;
